### ISSUE TITLE
Error with tetrate-istio-addon even with enable=false

### ISSUE THREAD

**Issue by ebertti** [aws-ia/terraform-aws-eks-blueprints#886]
Error inside blueprint module code, even without enabling *tetrate-istio-addon*

## Versions

- Module version [Required]: 
  - 4.7.0
- Terraform version: 
  - 1.2.7
- Provider version(s):
  + provider registry.terraform.io/gavinbunney/kubectl v1.14.0
  + provider registry.terraform.io/hashicorp/aws v4.27.0
  + provider registry.terraform.io/hashicorp/cloudinit v2.2.0
  + provider registry.terraform.io/hashicorp/helm v2.6.0
  + provider registry.terraform.io/hashicorp/kubernetes v2.12.1
  + provider registry.terraform.io/hashicorp/local v2.2.3
  + provider registry.terraform.io/hashicorp/null v3.1.1
  + provider registry.terraform.io/hashicorp/random v3.3.2
  + provider registry.terraform.io/hashicorp/time v0.8.0
  + provider registry.terraform.io/hashicorp/tls v3.4.0
  + provider registry.terraform.io/terraform-aws-modules/http v2.4.1

## Reproduction Code [Required]

```terraform


module "eks_blueprints" {
  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.7.0"

  cluster_name = local.cluster-name

  # EKS Cluster VPC and Subnet mandatory config
  vpc_id = var.context.vpc_id
  private_subnet_ids = var.context.subnets.privates

  # EKS CONTROL PLANE VARIABLES
  cluster_version = var.cluster_version


  cluster_security_group_tags = var.context.tags
  node_security_group_tags = var.context.tags


  # EKS MANAGED NODE GROUPS
  managed_node_groups = {
    default = {

      capacity_type = "SPOT"
      node_group_name = "${local.cluster-name}-default"
      instance_types = ["t3.large", "t3a.large", "t3.xlarge", "t3a.xlarge"]
      subnet_ids = var.context.subnets.privates

      min_size = 1
      max_size = 10
      desired_size = 1

      enable_monitoring = true
    }
  }

  tags = var.context.tags
}

module "eks_blueprints_kubernetes_addons" {
  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.7.0"

  eks_cluster_id = module.eks_blueprints.eks_cluster_id

  # EKS Addons
  enable_amazon_eks_vpc_cni = true
  enable_amazon_eks_coredns = true
  enable_amazon_eks_kube_proxy = true
  enable_amazon_eks_aws_ebs_csi_driver = true
  enable_tetrate_istio = false
  
}
```

```zsh
terraform plan 
```

## Expected behavior

Show result of plan 

## Actual behavior

Error o terraform code before plan

### Terminal Output Screenshot(s)

```
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 312, in module "tetrate_istio":
│  312:   distribution         = var.tetrate_istio_distribution
│ 
│ An argument named "distribution" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 313, in module "tetrate_istio":
│  313:   distribution_version = var.tetrate_istio_version
│ 
│ An argument named "distribution_version" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 314, in module "tetrate_istio":
│  314:   install_base         = var.tetrate_istio_install_base
│ 
│ An argument named "install_base" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 315, in module "tetrate_istio":
│  315:   install_cni          = var.tetrate_istio_install_cni
│ 
│ An argument named "install_cni" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 316, in module "tetrate_istio":
│  316:   install_istiod       = var.tetrate_istio_install_istiod
│ 
│ An argument named "install_istiod" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 317, in module "tetrate_istio":
│  317:   install_gateway      = var.tetrate_istio_install_gateway
│ 
│ An argument named "install_gateway" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 318, in module "tetrate_istio":
│  318:   base_helm_config     = var.tetrate_istio_base_helm_config
│ 
│ An argument named "base_helm_config" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 319, in module "tetrate_istio":
│  319:   cni_helm_config      = var.tetrate_istio_cni_helm_config
│ 
│ An argument named "cni_helm_config" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 320, in module "tetrate_istio":
│  320:   istiod_helm_config   = var.tetrate_istio_istiod_helm_config
│ 
│ An argument named "istiod_helm_config" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 321, in module "tetrate_istio":
│  321:   gateway_helm_config  = var.tetrate_istio_gateway_helm_config
│ 
│ An argument named "gateway_helm_config" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 322, in module "tetrate_istio":
│  322:   manage_via_gitops    = var.argocd_manage_add_ons
│ 
│ An argument named "manage_via_gitops" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/kubernetes.eks_blueprints_kubernetes_addons/modules/kubernetes-addons/main.tf line 323, in module "tetrate_istio":
│  323:   addon_context        = local.addon_context
│ 
│ An argument named "addon_context" is not expected here.
╵
ERRO[0016] 1 error occurred:
        * exit status 1

```

**Comment by ebertti** [comment]
I re-run from the beginning and works, i think it was a version error.

Sorry about it

closing this bug report

**Comment by iamyogeshg** [comment]
Hey @ebertti could you expand more on the resolution steps please? I am facing same issue here with same setup as yours.

**Comment by larry-aptos** [comment]
> Hey @ebertti could you expand more on the resolution steps please? I am facing same issue here with same setup as yours.

I resolved this my using the latest version 4.8.1 and the problem is gone.

**Comment by iamyogeshg** [comment]
> > Hey @ebertti could you expand more on the resolution steps please? I am facing same issue here with same setup as yours.
> 
> I resolved this by using the latest version 4.8.1 and the problem is gone.

Okay thanks for letting know. I had to update the TF to 1.2.8 to make it work.

**Comment by larry-aptos** [comment]
> > > Hey @ebertti could you expand more on the resolution steps please? I am facing same issue here with same setup as yours.
> > 
> > 
> > I resolved this by using the latest version 4.8.1 and the problem is gone.
> 
> Okay thanks for letting know. I had to update the TF to 1.2.8 to make it work.

Interesting; TF 1.2.4/1.2.7.1.2.8 doesn't work for 4.0.7. Module 4.7.0 doesn't work for me either...  4.8.1 is the only way for me to work.

Thanks!
